src/aggregate_with_randomness.cc

```diff
@@ -0,0 +1,247 @@
+                "aggregateWithRandomness")
+                .ThrowAsJavaScriptException();
+            return blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN;
+        }
+        Napi::Object set = set_value.As<Napi::Object>();
+
+        try {
+            blst_ts::PublicKey *to_aggregate = blst_ts::PublicKey::Unwrap(
+                set.Get("publicKey").As<Napi::Object>());
+            sets[i].pk = to_aggregate->point.get();
+        } catch (...) {
+            Napi::Error::New(
+                env,
+                "BLST_ERROR: Invalid PublicKey at index "s + std::to_string(i))
+                .ThrowAsJavaScriptException();
+            return blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN;
+        }
+
+        try {
+            Napi::Value val = set.Get("signature");
+            Napi::Uint8Array typed_array = val.As<Napi::Uint8Array>();
+            sets[i].sig_data = typed_array.Data();
+            sets[i].sig_length = typed_array.ByteLength();
+        } catch (...) {
+            Napi::Error::New(
+                env,
+                "BLST_ERROR: Invalid Signature at index "s + std::to_string(i))
+                .ThrowAsJavaScriptException();
+            return blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN;
+        }
+    }
+
+    return blst_ts::BLST_TS_ERROR::SUCCESS;
+}
+
+blst_ts::BLST_TS_ERROR aggregate_with_randomness(
+    blst::P1 &aggregate_key,
+    blst::P2 &aggregate_sig,
+    std::string &error_msg,
+    BlstTsAddon *module,
+    const std::vector<SignatureAndPublicKeySet> &sets,
+    const size_t &sets_length,
+    const bool &validate) {
+    for (uint32_t i = 0; i < sets_length; i++) {
+        blst::byte randomness[BLST_TS_RANDOM_BYTES_LENGTH];
+        if (!module->GetRandomNonZeroBytes(
+                randomness, BLST_TS_RANDOM_BYTES_LENGTH)) {
+            error_msg = "BLST_ERROR: Failed to generate random bytes";
+            return blst_ts::BLST_TS_ERROR::HAS_NATIVE_ERROR;
+        }
+
+        aggregate_key.add(
+            sets[i].pk->MultiplyBy(randomness, BLST_TS_RANDOM_BYTES_LENGTH));
+
+        blst::P2 sig;
+        try {
+            sig = blst::P2{sets[i].sig_data, sets[i].sig_length};
+        } catch (...) {
+            error_msg =
+                "BLST_ERROR: Invalid Signature at index "s + std::to_string(i);
+            return blst_ts::BLST_TS_ERROR::HAS_NATIVE_ERROR;
+        }
+
+        if (validate && (!sig.in_group() || sig.is_inf())) {
+            error_msg =
+                "BLST_ERROR: Invalid Signature at index "s + std::to_string(i);
+            return blst_ts::BLST_TS_ERROR::HAS_NATIVE_ERROR;
+        }
+
+        sig.mult(randomness, BLST_TS_RANDOM_BYTES_LENGTH * 8);
+        aggregate_sig.add(sig);
+    }
+
+    return blst_ts::BLST_TS_ERROR::SUCCESS;
+}
+
+Napi::Value AggregateWithRandomness(const Napi::CallbackInfo &info) {
+    BLST_TS_FUNCTION_PREAMBLE(info, env, module)
+    std::vector<SignatureAndPublicKeySet> sets{};
+    size_t sets_length{0};
```

Comment:
Why are they disjoint?

Reply:
Removed all references to sets_length now that I'm not using push_back to initialize the elements